lms/djangoapps/badges/models.py

```diff
@@ -0,0 +1,328 @@
+    )
+    default = models.BooleanField(
+        help_text=_(
+            u"Set this value to True if you want this image to be the default image for any course modes "
+            u"that do not have a specified badge image. You can have only one default image."
+        ),
+        default=False,
+    )
+
+    def __unicode__(self):
+        return u"<CourseCompleteImageConfiguration for '{mode}'{default}>".format(
+            mode=self.mode,
+            default=u" (default)" if self.default else u''
+        )
+
+    def clean(self):
+        """
+        Make sure there's not more than one default.
+        """
+        # pylint: disable=no-member
+        if self.default and CourseCompleteImageConfiguration.objects.filter(default=True).exclude(id=self.id):
+            raise ValidationError(_(u"There can be only one default image."))
+
+    @classmethod
+    def image_for_mode(cls, mode):
+        """
+        Get the image for a particular mode.
+        """
+        try:
+            return cls.objects.get(mode=mode).icon
+        except cls.DoesNotExist:
+            # Fall back to default, if there is one.
+            return cls.objects.get(default=True).icon
+
+    class Meta(object):
+        app_label = "badges"
+
+
+class CourseEventBadgesConfiguration(ConfigurationModel):
+    """
+    Determines the settings for meta course awards-- such as completing a certain
+    number of courses or enrolling in a certain number of them.
+    """
+    courses_completed = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"On each line, put the number of completed courses to award a badge for, a comma, and the slug of a "
+            u"badge class you have created that has the issuing component 'openedx__course'. "
+            u"For example: 3,enrolled_3_courses"
+        )
+    )
+    courses_enrolled = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"On each line, put the number of enrolled courses to award a badge for, a comma, and the slug of a "
+            u"badge class you have created that has the issuing component 'openedx__course'. "
+            u"For example: 3,enrolled_3_courses"
+        )
+    )
+    course_groups = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"Each line is a comma-separated list. The first item in each line is the slug of a badge class you "
+            u"have created that has an issuing component of 'openedx__course'. The remaining items in each line are "
+            u"the course keys the learner needs to complete to be awarded the badge. For example: "
+            u"slug_for_compsci_courses_group_badge,course-v1:CompSci+Course+First,course-v1:CompsSci+Course+Second"
+        )
+    )
+
+    def __unicode__(self):
+        return u"<CourseEventBadgesConfiguration ({})>".format(u"Enabled" if self.enabled else u"Disabled")
+
+    @staticmethod
+    def get_specs(text):
```

Comment:
Glad to see this is a staticmethod, but it really doesn't have anything to do at all with the model -- could it be moved to a utility library?